packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -112,10 +114,30 @@ contract OPContractsManager is ISemver {
         address mipsImpl;
     }
 
+    struct AddGameInput {
+        string saltMixer;
+        ISystemConfig systemConfig;
+        IProxyAdmin proxyAdmin;
+        GameType disputeGameType;
+        Claim disputeAbsolutePrestate;
+        uint256 disputeMaxGameDepth;
+        uint256 disputeSplitDepth;
+        Duration disputeClockExtension;
+        Duration disputeMaxClockDuration;
+        OutputRoot disputeStartingOutputRoot;
+        IBigStepper vm;
+        bool permissioned;
```

Comment:
Probably better to choose which contract to deploy based on the value of `disputeGameType` (permissioned if [type == 1](https://github.com/ethereum-optimism/optimism/blob/47957fbe53f9a229662e5843ffa0b968190aa18d/packages/contracts-bedrock/src/dispute/lib/Types.sol#L44-L44)).

Reply:
I don't think this is always true. Couldn't you in theory have a permissioned Asterisc game which would have a different dispute game type?

Reply:
You could, but I'm not sure why you would ever need / want more than 1 permissioned game type...

Reply:
It would be pretty bad if you meant to set a permissioned game and instead deployed a permissionless game - so however best to mitigate that.

Reply:
resolving given discord thread